app/server/appsmith-server/src/main/java/com/appsmith/server/services/LayoutActionServiceImpl.java

```diff
@@ -164,7 +164,18 @@ public LayoutActionServiceImpl(ActionService actionService,
                     if (action.getActionConfiguration() != null) {
                         newAction.setTimeoutInMillisecond(action.getActionConfiguration().getTimeoutInMillisecond());
                     }
-                    return newAction;
+
+                    // If the executeOnLoad field isn't true, set it to true
+                    if (!Boolean.TRUE.equals(action.getExecuteOnLoad())) {
+                        Action updateAction = new Action();
+                        updateAction.setExecuteOnLoad(true);
+
+                        return actionService.update(action.getId(), updateAction)
+                                .thenReturn(newAction);
```

Comment:
This will set the policies on the Action object to empty list no? Can we set it to null before passing to `.update`?

```suggestion
                        Action updateAction = new Action();
                        updateAction.setExecuteOnLoad(true);
                        updateAction.setPolicies(null);

                        return actionService.update(action.getId(), updateAction)
                                .thenReturn(newAction);
```

Reply:
The update function takes care of this by setting the policies to null in the base service. Not required here.